capirca/lib/nsxt.py

```diff
@@ -0,0 +1,339 @@
+  pass
+
+class UnsupportednsxtAccessListError(Error):
+  """Raised when we're give a non named access list."""
+  pass
+
+class NsxtAclTermError(Error):
+  """Raised when there is a problem in a nsxt access list."""
+  pass
+
+class NsxtDuplicateTermError(Error):
+  """Raised when there is a duplicate."""
+  pass
+
+class NsxtUnsupportedCriteriaOperator(Error):
+  """Raised when an unsupported criteria comparison operator is encountered."""
+  pass
+
+
+class Term(aclgenerator.Term):
+  """Creates a  single ACL Term for NSX-T."""
+
+  def __init__(self, term, filter_type, applied_to=None, af=4):
+    self.term = term
+    # Our caller should have already verified the address family.
+    assert af in (4, 6)
+    self.af = af
+    self.filter_type = filter_type
+    self.applied_to = applied_to
+
+  def __str__(self):
+    """Convert term to a rule string.
+
+    Returns:
+      A rule as a string.
+
+    """
+    # Verify platform specific terms. Skip whole term if platform does not
+    # match.
+    if self.term.platform:
+      if 'nsxt' not in self.term.platform:
+        return ''
+    if self.term.platform_exclude:
+      if 'nsxt' in self.term.platform_exclude:
+        return ''
+
+    term_keywords = self.term.__dict__
+    unsupported_keywords = []
+    for key in term_keywords:
+      if term_keywords[key]:
+        if ('translated' not in key) and (key not in _NSXT_SUPPORTED_KEYWORDS):
+              unsupported_keywords.append(key)
+    if unsupported_keywords:
+      logging.warning('WARNING: The keywords %s in Term %s are not supported '
+                      'in Nsxt ', unsupported_keywords, self.term.name)
+
+    name = self.term.name
+
+    notes = ''
+    if self.term.comment:
+      notes = self.term.comment
+
+    action = 'ALLOW'
+    if self.term.action:
+      action = self.term.action
+
+    source_address = 'ANY'
+    if self.term.source_address:
+      source_address = []
+      for i in self.term.source_address:
+        source_address.append(str(i))
+
+    destination_address = 'ANY'
+    if self.term.destination_address:
+      destination_address = []
+      for i in self.term.destination_address:
+        destination_address.append(str(i))
+
+    services = 'ANY'
+    # if self.term.services:
+    #   services = self.term.services
+
+    profiles = 'ANY'
+    # if self.term.profiles:
+    #   profiles = self.term.profiles
+
+    scope = 'ANY'
+    # if self.term.scope:
+    #   scope = self.term.scope
+
+    direction = 'IN_OUT'
+    # if self.term.direction:
+    #   direction = self.term.direction
+
+    tag = ''
+    # if self.term.tag:
+    #   tag = self.term.tag
+
+    ip_protocol = ''
+    # if self.term.ip_protocol:
+    #   ip_protocol = self.term.ip_protocol
+
+    rule =  {
+      "action": action,
+      "resource_type": "Rule",
+      "display_name": name,
+      "source_groups": source_address,
+      "destination_groups": destination_address,
+      "services": [
+          services
```

Comment:
Won't this _always_ use `"ANY"` as the only service in the list? Is that correct?

Same for `profiles` field, etc.

Reply:
left them with default values as those are mandatory for the API call and removed commented code